python/src/file.rs

```diff
@@ -403,9 +405,25 @@ impl LanceFileReader {
             .open_file(&path, &CachedFileSize::unknown())
             .await
             .infer_error()?;
+
+        let mut base_projection = None;
+        if let Some(columns) = columns {
+            let file_metadata = FileReader::read_all_metadata(&file).await
+                .map_err(|e| {
+                    PyIOError::new_err(format!("Error reading file metadata: {}", e))
+                })?;
+            base_projection = Some(ReaderProjection::from_column_names(
+                file_metadata.version(),
+                &file_metadata.file_schema,
+                &columns.iter().map(|s| s.as_str()).collect::<Vec<&str>>(),
+            ).map_err(|e| {
+                PyIOError::new_err(format!("Error creating projection: {}", e))
+            })?);
+        }
+
         let inner = FileReader::try_open(
```

Comment:
This will read the metadata twice.  Can you use `FileReader::try_open_with_file_metadata` if there is a projection?

You could also _always_ call `FileReader::read_all_metadata` (even if there is no projection) and use `try_open_with_file_metadata` which might be slightly more DRY.